### ISSUE TITLE
deployDoc cannot handle pdf-generating Rmd

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rsconnect#36]
This is perhaps part rsconnect, part rmarkdown.

Create an Rmd with the header YAML:

```
output: pdf_document
```

Attempt to deploy that document and you get an error:

```
> rsconnect::deployDoc("pdf.Rmd",account="dev")
Discovering document dependencies... 
 Show Traceback
 
 Rerun with Debug
 Error in (function (toc = FALSE, toc_depth = 2, number_sections = FALSE,  : 
  unused argument (self_contained = FALSE) 
```

This is because `rmarkdown::find_external_resources` eventually calls `rmarkdown::render` with `self_contained=FALSE`. This option is not accepted if `output: pdf_document` (see https://github.com/rstudio/rmarkdown/issues/437).

**Comment by aronatkins** [comment]
I am working around this by manually calling `deployApp` as follows:

```
rsconnect::deployApp(appDir=".", appFiles=c("pdf.Rmd", "shoe_avatar.jpg"), appPrimaryDoc="pdf.Rmd",account="dev")
```

**Comment by jjallaire** [comment]
Yikes! @jmcphers Should we only be calling find_external_resources for HTML based formats?

**Comment by aronatkins** [comment]
It's possible to embed on-disk images in PDF output; can `find_external_resources` handle other output formats?

**Comment by jmcphers** [comment]
There's definitely some undefined behavior here because we haven't really thought through what happens when you publish this kind of asset. There are a few ways we could proceed:

a) Suppress these assets entirely

- Hide Publish button in IDE if default output format is not known to produce HTML 
- Have `deployDoc` introspect the YAML header and throw up its hands if default output format is PDF
- Have `find_external_resources` do the same 

b) Permit publishing documents with PDF as the default output format as long as they contain an HTML output format, too

- As (a), but attempt to find an HTML-generating format, and publish as though that format were the default
- Do work on the connect server to choose and render an HTML-based format even if PDF is the default

c) Fully support publishing Rmds that can create PDFs

- Implement PDF resource discovery in `find_external_resources` (note that this might involve parsing LaTeX) 
- Implement PDF rendering/serving in Connect
- Possibly implement PDF report e-mailing in Connect

d) Support publishing both Rmds that can create PDFs and PDFs themselves

- As (c) but also put a Publish button on our internal PDF viewer, and add support for PDFs as a new static content type

For the alpha I'm okay with (a) but if either if you feel strongly we could use this as an impetus to do some feature implementation a la b, c, or d.

**Comment by jjallaire** [comment]
>
> c) Fully support publishing Rmds that can create PDFs
>
>    - Implement PDF resource discovery in find_external_resources (note
>    that this might involve parsing LaTeX)
>    - Implement PDF rendering/serving in Connect
>    - Possibly implement PDF report e-mailing in Connect
>
> d) Support publishing both Rmds that can create PDFs and PDFs themselves
>
>    - As (c) but also put a Publish button on our internal PDF viewer, and
>    add support for PDFs as a new static content type
>
>
In the last Connect standup we discussed being able to render reports as
PDFs (since HTML reports are poor attachments we think that PDF attachments
will be very popular). I therefore think that some level of PDF-savvy on
the client is in order. I'd actually favor doing (d) as I think the work
will be pretty minimal and mostly on the IDE side.

My thinking is that we can do a limited but I think 95/5 version of PDF
resource discovery with 2 things: (a) Pickup the .tex files referenced in
the 'includes' parameter (if any); and (b) Use the embedded markdown parser
we already have in the IDE (sundown) to pick out image references.

Since our publishing pipeline also allows for users to whitelist files into
the list external resources I think a mostly accurate method is okay.

Feel free to push back if you think I'm being too optimistic!

Jonathan, let's discuss what we think this would take on Monday (I'm at
R/Finance tomorrow).

**Comment by aronatkins** [comment]
Related: https://github.com/rstudio/rmarkdown/issues/438

**Comment by aronatkins** [comment]
Fixed with #38 plus a variety of rmarkdown changes.
